src/rdkafka.c

```diff
@@ -2330,6 +2342,24 @@ rd_kafka_t *rd_kafka_new(rd_kafka_type_t type,
                 rk->rk_conf.security_protocol = RD_KAFKA_PROTO_PLAINTEXT;
         }
 
+        /* Create background thread and queue if background_event_cb()
```

Comment:
Since rd_kafka_sasl_oauthbearer_init(rk) called rd_kafka_sasl_background_callbacks_enable() already, it will create a new background thread.  if this part is after calling rd_kafka_sasl_oauthbearer_init(), it will fail to create background thread.

Reply:
Huhm okay, maybe it is better to leave the original order of the code and instead check to see that the background has not been created (check rk->rk_background.q while holding the wrlock)